.travis.yml

```diff
@@ -21,10 +24,10 @@ install:
 
 # Generate a self-signed X.509 certificate for TLS.
 before_script:
- - openssl req -x509 -newkey rsa:4096 -keyout server.key -out server.crt -days 365 -nodes -subj /CN=localhost
+ - if [[ "$TEST_SUITE" == "integ-test" ]]; then openssl req -x509 -newkey rsa:512 -keyout server.key -out server.crt -days 365 -nodes -subj /CN=localhost; fi
```

Comment:
is there a reason not to do these in `travis-test.sh`?

Reply:
We can move the openssl, but I find it nice to install the dependencies before we start the tests. It isn't the best reason in the world.

Reply:
I feel like I'd rather have all the magic in travis-test rather than split between that script and this config file.